evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -724,7 +729,9 @@ pub fn apply(
             serialize_json(state, &inner.0, &inner.1, &array_str_to_str(&inner.2))
         }
         HEVMCalls::WriteJson0(inner) => write_json(state, &inner.0, &inner.1, None),
-        HEVMCalls::WriteJson1(inner) => write_json(state, &inner.0, &inner.1, Some(&inner.2)),
+        HEVMCalls::WriteJson1(inner) => {
+            write_json(state, &inner.0, &inner.1, Some(&format!("$.{}", &inner.2)))
```

Comment:
I don't understand why we had to remove support for `.key` notation? Is this unavoidable? This is a pretty big breaking change and I think also means we're using a different syntax than standard `jq`/JSONPath syntax? My understanding is that you need the `.`, but I could be wrong. See below where `jq` fails to parse without it

<img width="514" alt="image" src="https://user-images.githubusercontent.com/17163988/216987398-d6349070-850d-49ca-a13a-f38fdb1d3d44.png">

Reply:
The idea was to make it easier for users, but I guess you are right.

We could remove any formatting and force users to strictly abide (`$.key`), with $ meaning the root of the json object.

Reply:
Sorry wdym exactly, would the breaking change instead be to force users to have the `$`? 

I think my current confusion is that my understanding is the issue was with ABI encoding the result when there are multiple returned values, so why does that affect the syntax used to specify keys?

I'd have though the only change needed here was this part:

```rust
let abi_encoded = if res.len() == 1 {
    abi::encode(&[Token::Bytes(abi::encode(&res))])
} else {
    abi::encode(&[Token::Bytes(abi::encode(&[Token::Array(res)]))])
};
```

Reply:
With the bug, the code would *work* with both `.key` and `key,` due to the bug. 

Now, we have to make a choice. If we want to be strict with jsonPath, I **think** we should support `$.key` and not `.key`, even if it's breaking change, because `$` signifies the root of the json object. So, the cheatcode internally doesn't modify the key in any way. If we support `.key`, the cheatcode internally modifies it to `$.key`.

Reply:
Hmm, I see. Supporting `key` originally feels like a bug, cause I don't think that's really valid syntax anywhere else, so I'm ok with dropping that. Then changing `parseJson` to support both `$.key` AND `.key` (by automatically prepending a `$` in the latter case) seems ideal, that way we minimize breaking changes and people can use familiar jq-syntax.

Reply:
I think I would prefer to go with `.key`,  assuming that there aren't cases where not starting from the root is preferable. From what I found on the internet, all examples start with `$`.

Reply:
Just to be clear, did this PR break anyone using `.key` to access a field of a json object? If yes we should bring it back.

Reply:
@gakonst it was fixed in #4288